### PR TITLE
Tag filtering function

### PR DESCRIPTION
**Pull request by ryomarugon**
## Summary
Create tag filtering function

## Changes 

- Added v-if, v-else to TaskList.vue to change display of tasks
- Changed the the name of tags filtered in TagFilter.vue from "selectedTags" to "filteredTags " to distinguish from the one 
 in Modal.vue
- Added filterTags to TodoApp.vue and wrote codes for filtering  

## Related File
- TagFilter.vue
- TaskList.vue
- TodoApp.vue

### CODE REVIEW

**Comment by aemanzano** [comment]
Filter is working but drag and drop is not working when you started filtering.
Maybe as Maki-san if you can't drag and drop if you selected a filter.
https://github.com/ryomarugon/TodoApp/assets/95004782/cc2e9ba8-466e-4220-aab9-0a1f6e7b7abd

**Comment by ryomarugon** [comment]
Probably I could fix it by the change above

**Comment by aemanzano** [comment]
1. Since this is for calculating, transfer this to computed
image

2. Wrap all elements within a div before the template
image
image

3. Make classes consistent. Some are "-" but some are "_"
image

4. Be consistent when naming emit functions. Some are camel case some are using ":"
image

File: components/TodoApp.vue [hunk 1]
```diff
@@ -1,39 +1,52 @@
...
+    <Modal
+      @closeModal="closeModal"
+      v-if="showModal"
+      :status="selectedStatus"
+      :index="taskIndex"
+      :tagHistory="tagHistory"
+      :showModal="true"
+      @addTask="addTask"
+    ></Modal>
+  </div>
```
Comment (aemanzano) [inline components/TodoApp.vue:49]: Put all elements inside a div before the template

**Pull request merged**